unison-cli/src/Unison/Codebase/Editor/HandleInput.hs

```diff
@@ -934,19 +935,21 @@ loop e = do
               #numberedArgs .= fmap Name.toString patches
             FindShallowI pathArg -> do
               Cli.Env {codebase} <- ask
-              sbhLength <- liftIO (Codebase.branchHashLength codebase)
-              rootBranch <- Cli.getRootBranch
 
               pathArgAbs <- Cli.resolvePath' pathArg
               entries <- liftIO (Backend.lsAtPath codebase Nothing pathArgAbs)
               -- caching the result as an absolute path, for easier jumping around
               #numberedArgs .= fmap entryToHQString entries
-              let ppe =
-                    Backend.basicSuffixifiedNames
-                      sbhLength
-                      rootBranch
-                      (Backend.AllNames (Path.unabsolute pathArgAbs))
-              Cli.respond $ ListShallow ppe entries
+              getRoot <- atomically . STM.readTMVar <$> use #root
+              let buildPPE = do
```

Comment:
We only use the PPE when pretty-printing type signatures in LS output; but the root branch and most branches on the way to wherever the user is working won't have types in them.

Deferring loading the PPE unless we know it's necessary means the user can `ls` and `cd` their way down to their project without waiting for the root to load, and when they get there, most operations will be self-contained and we don't need the root branch anyways.